### PR TITLE
fix(laravel): fix hiding/showing relationships

### LINKED ISSUE #6678: [Laravel]: Relationships are always exposed even if set in `hidden` (or not set in `visible`)
**API Platform version(s) affected**: 4.0.1

**Description**  
If you have a Model with a relationship defined, that relationship will always be visible / exposed, even if you define `hidden` or `visible` on the model

**How to reproduce**  
Add relationship to a Laravel Model (for example `roles`) and put it inside the `$hidden` array on your model.

### PR DESCRIPTION
**Pull request by toitzi**
Check if is in visible or hidden (if exists) just like with regular proeprties

Closes: #6678

| Q             | A
| ------------- | ---
| Branch?       | 4.0
| Tickets       | Closes #6678 
| License       | MIT
| Doc PR        |

### CODE REVIEW

**Review by dunglas — APPROVED** [review]
Thanks! Could you add a test, please?

**Comment by toitzi** [comment]
Sure im on it

**Comment by toitzi** [comment]
@dunglas Tests are added

File: tests/Laravel/Metadata/VisibleHiddenAttributesTest.php [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+    {
+        $model = new class extends Model {};
+
+        $result = $this->invokePrivateMethod($this->modelMetadata, 'attributeIsHidden', ['any_attribute', $model]);
+        $this->assertFalse($result);
+    }
+
+    public function testGetRelationsReturnsCorrectRelations(): void
+    {
+        $relation = \Mockery::mock(HasMany::class);
```
Comment (soyuka) [inline tests/Laravel/Metadata/VisibleHiddenAttributesTest.php:76]: We don't use Mockery on our code base, maybe that a functional test would've been enough for this use case? If you want to unit test this, you may test the result of `getRelations`.

File: tests/Laravel/Metadata/VisibleHiddenAttributesTest.php [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+use ApiPlatform\Laravel\Eloquent\Metadata\ModelMetadata;
+use Illuminate\Database\Eloquent\Model;
+use Illuminate\Database\Eloquent\Relations\HasMany;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @author Tobias Oitzinger <tobiasoitzinger@gmail.com>
+ */
+class VisibleHiddenAttributesTest extends TestCase
```
Comment (soyuka) [inline tests/Laravel/Metadata/VisibleHiddenAttributesTest.php:24]:
```suggestion
class ModelMetadataTest extends TestCase
```

**Comment by soyuka** [comment]
took the liberty to change the test a bit, thanks @toitzi !

**Pull request merged**